gui/wxpython/startup/guiutils.py

```diff
@@ -692,6 +697,58 @@ def delete_grassdb_interactively(guiparent, grassdb):
     return deleted
 
 
+def can_switch_mapset_interactively(guiparent, grassdb, location, mapset):
+    """
+    Checks if mapset is locked and offers to remove the lock file.
+
+    Returns True if user wants to switch to the selected mapset in spite of
+    removing lock. Returns False if a user wants to stay in the current
+    mapset or if an error was encountered.
+    """
+    can_switch = True
+    lockfile = get_lockfile_if_present(grassdb, location, mapset)
+
+    if lockfile:
+        user = get_gislock_owner(lockfile)
+        timestamp = (datetime.datetime.fromtimestamp(
+            os.path.getmtime(lockfile))).replace(microsecond=0)
+        dlg = wx.MessageDialog(
+            parent=guiparent,
+            message=_("User {user} is already running GRASS in selected mapset "
+                      "<{mapset}>\n (file {lockfile} created {timestamp} "
+                      "found).\n\n"
+                      "Concurrent use not allowed.\n\n"
+                      "Do you want to stay in the current mapset or remove "
+                      ".gislock and switch to selected mapset?"
+                      ).format(user=user,
+                               mapset=mapset,
+                               lockfile=lockfile,
+                               timestamp=timestamp),
+            caption=_("Lock file found"),
+            style=wx.YES_NO | wx.NO_DEFAULT | wx.ICON_QUESTION,
+        )
+        dlg.SetYesNoLabels("&Switch to selected mapset",
+                           "&Stay in current mapset")
```

Comment:
The `&` is what is used to identify a character which will be used with Alt key to "press" the button without using a mouse pointer. They need to be at least different, but I would probably go with `"S&witch...` and `"S&tay...`.